src/main/java/hudson/plugins/gradle/injection/MavenExtensionsHandler.java

```diff
@@ -0,0 +1,90 @@
+package hudson.plugins.gradle.injection;
+
+import hudson.FilePath;
+import org.apache.commons.io.IOUtils;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+
+import static hudson.plugins.gradle.injection.CopyUtil.copyResourceToNode;
+import static hudson.plugins.gradle.injection.MavenExtensionsHandler.MavenExtension.CCUD;
+import static hudson.plugins.gradle.injection.MavenExtensionsHandler.MavenExtension.GRADLE_ENTERPRISE;
+
+public class MavenExtensionsHandler {
+
+    static final String LIB_DIR_PATH = "jenkins-gradle-plugin/lib";
+
+    private final MavenExtensionFileHandler geExtensionHandler = new MavenExtensionFileHandler(GRADLE_ENTERPRISE);
+    private final MavenExtensionFileHandler ccudExtensionHandler = new MavenExtensionFileHandler(CCUD);
+
+    public void copyGradleEnterpriseExtensionToAgent(FilePath rootPath) throws IOException, InterruptedException {
+        geExtensionHandler.copyExtensionToAgent(rootPath);
+    }
+
+    public FilePath getGradleEnterpriseExtensionPath(FilePath rootPath) throws IOException {
+        return geExtensionHandler.getAgentExtensionPath(rootPath);
+    }
+
+    public void copyCCUDExtensionToAgent(FilePath rootPath) throws IOException, InterruptedException {
```

Comment:
```suggestion
    public void copyCcudExtensionToAgent(FilePath rootPath) throws IOException, InterruptedException {
```

Reply:
I would keep it all uppercase because CCUD is already an acronym not a word.